@xen-orchestra/xapi/vm.mjs

```diff
@@ -599,6 +599,32 @@ class Vm {
     }
   }
 
+  async leafCoalesce($defer, vmRef) {
+    const vmUuid = await this.getField('VM', vmRef, 'uuid')
+    try {
+      await this.callAsync('VM.suspend', vmRef)
+      $defer(() => this.resumeVm(vmUuid))
+    } catch (error) {
+      if (error.code !== 'VM_BAD_POWER_STATE') {
+        throw error
+      }
+
+      const powerState = error.params[2].toLowerCase()
+      if (powerState !== 'halted' && powerState !== 'suspended') {
+        throw error
+      }
+    }
+
+    // result can be: `Success` or `VM has no leaf-coalesceable VDIs`
+    const result = await this.callAsync('host.call_plugin', this.pool.master, 'coalesce-leaf', 'leaf-coalesce', {
```

Comment:
Add a link to the plugin's documentation here.

Reply:
You can keep your link because it's relevant

But I was talking about this one: https://docs.xenserver.com/en-us/xenserver/8/storage/manage.html#reclaim-space-by-using-the-offline-coalesce-tool